### PR TITLE
feat: use env instead of args in docker images

### PR DESCRIPTION
**Pull request by onedr0p**
## Description

This sets up the defaults in the docker container to use config from env instead of args

## Related Issue

As discussed here https://github.com/jesec/flood/pull/261

## Types of changes

- [ ] Breaking change (changes that break backward compatibility of public API or CLI - semver MAJOR)
- [ ] New feature (non-breaking change which adds functionality - semver MINOR)
- [ ] Bug fix (non-breaking change which fixes an issue - semver PATCH)

### CODE REVIEW

**Comment by jesec** [comment]
I think we should keep the args way in the doc, while suggesting env as an alternative.

**Comment by jesec** [comment]
And, can we use multiple ENV and move options near the entry point?

File: distribution/containers/Dockerfile.distroless [hunk 1]
```diff
@@ -42,5 +49,15 @@ FROM flood as rtorrent-flood
...
+ENV \
+    HOME=/home/download \
+    FLOOD_OPTION_HOST="0.0.0.0" \
+    FLOOD_OPTION_PORT="3000" \
+    FLOOD_OPTION_RTORRENT="true"
+
 # Flood with managed rTorrent daemon
-ENTRYPOINT ["flood", "--host=0.0.0.0", "--rtorrent"]
+ENTRYPOINT ["/sbin/tini", "--"]
+CMD ["flood"]
```
Comment (jesec) [inline distribution/containers/Dockerfile.distroless:63]: we still probably want to use `flood` as the entrypoint.

Comment (onedr0p) [reply distribution/containers/Dockerfile.distroless]: Updated

**Comment by onedr0p** [comment]
> And, can we use multiple ENV and move options near the entry point?

Do you really want separate docker layers for each ENV? Currently it is treated as 1 layer.

**Comment by jesec** [comment]
> > And, can we use multiple ENV and move options near the entry point?
> 
> Do you really want separate docker layers for each ENV? Currently it is treated as 1 layer.

It actually makes more sense, as each ENV is not related to others. I am OK with zero-size layers, and that looks better in the Dockerfile.

**Comment by jesec** [comment]
the wording looks a bit weird to me. I will probably prefer to keep the original one and add something like "Alternatively, you may use env ...".

**Comment by onedr0p** [comment]
Now that env is default in the dockerfiles you still want to point people to using args over env?

**Comment by jesec** [comment]
> Now that env is default in the dockerfiles you still want to point people to using args over env?

well. unless you think `docker ... -e FLOOD_OPTION_XXX=true ...` is more convenient than `docker ... --xxx`. 

It is OK to use env in chart or other file-based config. But for the command line, args is still the more user-friendly one.

**Comment by onedr0p** [comment]
> unless you think `docker ... -e FLOOD_OPTION_XXX=true ...` is more convenient than `docker ... --xxx`.

For the complete docker beginner, no otherwise I do.

> It is OK to use env in chart or other file-based config. But for the command line, args is still the more user-friendly one.

But I don't think a lot of people are using `docker run` to run rtorrent, it's more more likely they are using docker-compose in which any method is fine but env is more widely used.

**Comment by onedr0p** [comment]
But you are right we are describing this with `docker run` I'll update it

**Comment by jesec** [comment]
Beginners exist and there are probably a lot. They probably don't know how to use Docker Compose. After all, one can have a poor man's service up with simple `nohup docker run ... &` in any shell.

File: distribution/containers/Dockerfile.release [hunk 2]
```diff
@@ -21,8 +21,12 @@ RUN adduser -h /home/download -s /sbin/nologin --disabled-password download
 # Run as "download" user
 USER download
 
+ENV HOME=/home/download
```
Comment (jesec) [inline distribution/containers/Dockerfile.release:24]: For non-distroless (normal) images, it is unnecessary (and harmful) to override HOME inside the Dockerfile.

Comment (jesec) [reply distribution/containers/Dockerfile.release:24]: as there is one by default.

Comment (onedr0p) [reply distribution/containers/Dockerfile.release]: Updated

**Comment by onedr0p** [comment]
Readme was updated

**Comment by jesec** [comment]
`To view more available environment options see [here](https://github.com/jesec/flood/blob/master/config.ts).` 

Note that `FLOOD_OPTION_*` is simply env way to set a CLI argument. So `--help` is probably a more straightforward way to see help texts. Or run it with `FLOOD_OPTION_HELP=true` (but that's weird). And, there is already a section about `Configuration`: https://github.com/jesec/flood#configuration .

**Comment by onedr0p** [comment]
Updated the link to point to https://github.com/jesec/flood#configuration

**Comment by jesec** [comment]
I made some changes. Please check it out.

**Comment by onedr0p** [comment]
Good catch on tini-static for distroless. Looks good

LGTM!

**Pull request merged**